### ISSUE TITLE
[bug] global angular-cli install produces peer warnings

### ISSUE THREAD

**Issue by amitport** [angular/angular-cli#9200]
installing `@angular/cli@1.6.4` *globally* produces peer warnings:

running: `npm i @angular/cli@1.6.4 -g` 
produces:
`npm WARN @schematics/angular@0.1.13 requires a peer of @angular-devkit/schematics@0.0.46 but none was installed.`

`npm WARN @schematics/schematics@0.0.13 requires a peer of @angular-devkit/schematics@0.0.46 but none was installed.`

**Comment by tatianajiselle** [comment]
Getting the same problem here.

**Comment by mkotsollaris** [comment]
Same here.

**Comment by OrionStark** [comment]
Same here. Is it okay?

**Comment by tatianajiselle** [comment]
havent encountered any errors from it ATM as im developing, but would be nice to get a fix/resolved.

**Comment by mkotsollaris** [comment]
@tatianajiselle when I run `ng -v` it gives me a command not found. Are you getting this?

**Comment by tatianajiselle** [comment]
@mkotsollaris no im not, thats strange:
my output:

  _                      _                 ____ _     ___
   / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
  / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
 / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
/_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
               |___/
@angular/cli: 1.4.2
node: 8.9.4
os: darwin x64
@angular/animations: 4.3.1
@angular/common: 4.3.1
@angular/compiler: 4.3.1
@angular/core: 4.3.1
@angular/forms: 4.3.1
@angular/http: 4.3.1
@angular/platform-browser: 4.3.1
@angular/platform-browser-dynamic: 4.3.1
@angular/router: 4.3.1
@angular/cli: 1.4.2
@angular/compiler-cli: 4.3.1
@angular/language-service: 4.3.1
typescript: 2.3.3

**Comment by tatianajiselle** [comment]
npm install --save @angular-devkit/schematics@0.0.46

**Comment by MiguelMunoz** [comment]
Okay, I'm new to all this, including npm, but here's what I did. 
I got the same errors: 
```
npm WARN @schematics/angular@0.1.13 requires a peer of @angular-devkit/schematics@0.0.46 but none was installed.
npm WARN @schematics/schematics@0.0.13 requires a peer of @angular-devkit/schematics@0.0.46 but none was installed.
```

Then I did an `npm init` and `npm install`, the I typed in tatianajiselle suggested command: `npm install --save @angular-devkit/schematics@0.0.46`

That gave me the same warnings. Then I typed ng -v, and got this response:

```
Unable to find "@angular/cli" in devDependencies.

Please take the following steps to avoid issues:
"npm install --save-dev @angular/cli@latest" 


    _                      _                 ____ _     ___
   / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
  / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
 / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
/_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
               |___/
    
Angular CLI: 1.6.4
Node: 6.9.4
OS: darwin x64
Angular: undefined
... 


@angular-devkit/core: 0.0.25
@angular-devkit/schematics: 0.0.46
@ngtools/json-schema: 1.1.0
@schematics/schematics: 0.0.13
typescript: error
webpack: error
``` 

I took its suggestion and typed `npm install --save-dev @angular/cli@latest`

I think that worked. When I type `ng -v` I get this:

```
Angular CLI: 1.6.4
Node: 6.9.4
OS: darwin x64
Angular: undefined
... 

@angular/cli: 1.6.4
@angular-devkit/build-optimizer: 0.0.38
@angular-devkit/core: 0.0.25
@angular-devkit/schematics: 0.0.46
@ngtools/json-schema: 1.1.0
@ngtools/webpack: 1.9.4
@schematics/angular: 0.1.13
@schematics/schematics: 0.0.13
typescript: 2.6.2
webpack: 3.10.0
```

**Comment by bmercier-manakin** [comment]
I got same error btw,
I've upgraded my project today and here it is:
```warning "@angular/cli > uglifyjs-webpack-plugin > schema-utils > ajv-keywords@2.1.1" has unmet peer dependency "ajv@^5.0.0".
warning "@angular/cli > @angular-devkit/schematics > @schematics/schematics@0.0.13" has incorrect peer dependency "@angular-devkit/schematics@0.0.46".
warning "@angular/cli > @schematics/angular@0.1.13" has incorrect peer dependency "@angular-devkit/schematics@0.0.46".
```

I guess they are some version hard written somewhere, no ?

```
Angular CLI: 1.6.4
Node: 8.9.1
OS: linux x64
Angular: 5.2.0
... animations, common, compiler, compiler-cli, core, forms
... http, platform-browser, platform-browser-dynamic
... platform-server, router

@angular/cdk: 5.0.4
@angular/cli: 1.6.4
@angular/flex-layout: 2.0.0-beta.12
@angular/language-service: 4.4.3
@angular/material: 5.0.4
@angular-devkit/build-optimizer: 0.0.38
@angular-devkit/core: 0.0.25
@angular-devkit/schematics: 0.0.48
@ngtools/json-schema: 1.1.0
@ngtools/webpack: 1.9.4
@schematics/angular: 0.1.13
@schematics/schematics: 0.0.13
typescript: 2.4.2
webpack: 3.10.0
```

**Comment by joshuwhi** [comment]
I get the same error aswell.

Also, when I tried the same steps as @MiguelMunoz everything appears fine:
```
ng -v                                                                        

    _                      _                 ____ _     ___
   / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
  / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
 / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
/_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
               |___/

Angular CLI: 1.6.4
Node: 8.9.4
OS: linux x64
Angular: 5.2.0
... animations, common, compiler, compiler-cli, core, forms
... http, language-service, platform-browser
... platform-browser-dynamic, router

@angular/cli: 1.6.4
@angular-devkit/build-optimizer: 0.0.38
@angular-devkit/core: 0.0.25
@angular-devkit/schematics: 0.0.46
@ngtools/json-schema: 1.1.0
@ngtools/webpack: 1.9.4
@schematics/angular: 0.1.13
@schematics/schematics: 0.0.13
typescript: 2.5.3
webpack: 3.10.0
```


but then using `ng generate module foo` gives me the error: 
```
Error: dryRunSink.commit(...).ignoreElements is not a function
dryRunSink.commit(...).ignoreElements is not a function
```

**Comment by tatianajiselle** [comment]
You can try upgrading all your dependencies and versions to latest. Deleting the folder node_modules and then running npm install

**Comment by joshuwhi** [comment]
@tatianajiselle 
 - pureged all node_modules globally

 - installed @angular/cli@latest (same @schematics warning appears)

- generated a new project and get the error: 
```
Unable to find "@angular/cli" in devDependencies.

Please take the following steps to avoid issues:
"npm install --save-dev @angular/cli@latest"
```

- ran `npm install --save-dev @angular/cli@latest`

- wiped out node_modules folder

-ran `npm install`

No change after all these steps. Still have the same issues I stated above in a fresh project

node: v8.9.4
npm: 5.6.0

**Comment by granteagon** [comment]
Also seeing this issue
```Angular CLI: 1.5.5
Node: 8.9.1
OS: win32 x64
Angular: 5.0.1
... animations, common, compiler, core, forms, http
... platform-browser, platform-browser-dynamic, router, upgrade

@angular/cli: 1.5.5
@angular/compiler-cli: 5.0.0
@angular/language-service: 5.0.0
@angular-devkit/build-optimizer: 0.0.38
@angular-devkit/core: 0.0.25
@angular-devkit/schematics: 0.0.46
@ngtools/json-schema: 1.1.0
@ngtools/webpack: 1.8.5
@schematics/angular: 0.1.13
@schematics/schematics: 0.0.13
typescript: 2.4.2
webpack: 3.8.1
```

**Comment by hattifattener** [comment]
+1
Ubuntu 17.10
Node v6.11.4
NPM 3.5.2

> npm i -g @angular/cli
.....
/usr/local/lib
└─┬ @angular/cli@1.6.4 
  └─┬ UNMET PEER DEPENDENCY @angular-devkit/schematics@0.0.48
    └── UNMET PEER DEPENDENCY @angular-devkit/schematics@0.0.46

npm ERR! peer dep missing: @angular-devkit/schematics@0.0.46, required by @schematics/angular@0.1.13

**Comment by lucasvst** [comment]
+1
same here :(

**Comment by granteagon** [comment]
Fixed it.

PowerShell run as Administrator:
```
> rm node_modules
> npm cache verify
> npm install
> npm install @angular-devkit/schematics@0.0.45 --no-save
```
Cobbled this solution together from #9194

Can now generate stuff again.

**Comment by andrewmmc** [comment]
Same here.
npm v.5.6.0
node v.6.11.2
macOS v.10.13.2

```
/usr/local/lib
├── @angular/cli@1.6.4

npm ERR! peer dep missing: @angular-devkit/schematics@0.0.46, required by @schematics/angular@0.1.13
```

**Comment by hansl** [comment]
Yes, we had some issues with 1.6.4. None of those affected usage directly, but there were warnings when globally installing (locally was fine).

These have been resolved in 1.6.5. Update your packages accordingly. Thanks.
